frontend/packages/console-shared/src/hooks/useResizeObserver.ts

```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
+
+export const useResizeObserver = (
+  callback: ResizeObserverCallback,
+  targetElement: HTMLElement | null = document.querySelector('#app'),
+  observerOptions: ResizeObserverObserveOptions = undefined,
+): void => {
+  const targetElementRef = React.useRef(targetElement);
+  targetElementRef.current = targetElement;
```

Comment:
No need for a `ref` as the target element should be a dependency on the effect hook.